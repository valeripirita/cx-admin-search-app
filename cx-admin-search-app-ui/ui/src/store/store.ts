import { configureStore, combineReducers } from '@reduxjs/toolkit';
import searchFormReducer from './slices/searchFormSlice';
import { accountServiceApi } from './apis/accountServiceApi';

const rootReducer = combineReducers({
    searchFormReducer,
    [accountServiceApi.reducerPath]: accountServiceApi.reducer
});

export const setupStore = () => configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware()
        .concat(accountServiceApi.middleware)
})

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore['dispatch'];
