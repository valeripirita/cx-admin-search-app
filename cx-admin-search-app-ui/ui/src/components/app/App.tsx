import '../../../styles.css'

import React from 'react';
import { Provider } from 'react-redux';

import { toRelativeUrl } from '@okta/okta-auth-js';
import { Security } from '@okta/okta-react';

import { oktaConfig } from '../../config';
import { setupStore } from '../../store/store';
import Main from '../main/Main';
const store = setupStore();

const App = () => {

    const restoreOriginalUri = async (_oktaAuth: any, originalUri: string) => {
        { to: (toRelativeUrl(originalUri || '/', window.location.origin)) }
    };

    return (
        <Security oktaAuth={oktaConfig} restoreOriginalUri={restoreOriginalUri}>
            <Provider store={store}>
                <Main />
            </Provider>
        </Security>
    );
};

export default App;
