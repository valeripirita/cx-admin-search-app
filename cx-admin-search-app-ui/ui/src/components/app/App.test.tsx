import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { afterEach, describe, expect, it } from '@jest/globals';
import { cleanup } from '@testing-library/react';
import App from './App';
import { setupStore } from '../../store/store';

describe('App', () => {

    afterEach(cleanup)

    it('should render', () => {

        const store = setupStore();

        const { asFragment, unmount } = render(<Provider store={store}><App /></Provider>);

        expect(asFragment()).toMatchSnapshot();

        unmount()
    });
})