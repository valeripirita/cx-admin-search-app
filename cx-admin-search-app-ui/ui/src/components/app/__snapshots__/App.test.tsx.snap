// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<DocumentFragment>
  <main
    class="bg-neutral-100 flex flex-row h-full w-full"
  >
    <section
      class="box-border w-[298px] shadow bg-white border-2 border-primary-500 rounded-tr px-4 pb-4 h-full overflow-auto"
    >
      <form
        class="text-sm"
        data-test="search-form"
      >
        <fieldset
          class="mb-2 w-full"
        >
          <label
            class="-mt-4.5 relative cursor-pointer"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-stone-600"
            >
              Jurisdiction
            </p>
            <select
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full appearance-none cursor-pointer"
              name="jurisdiction"
            >
              <option
                value="EU"
              >
                Europe
              </option>
              <option
                value="NJ"
              >
                New Jersey
              </option>
              <option
                value="NY"
              >
                New York
              </option>
              <option
                value="AZ"
              >
                Arizona
              </option>
              <option
                value="ON"
              >
                Ontario
              </option>
            </select>
            <svg
              class="absolute right-2 top-1/2 translate-y-0.5 pointer-events-none"
              data-file-name="SvgChevronDown"
            />
          </label>
        </fieldset>
        <fieldset
          class="mb-4 w-full"
        >
          <hr
            class="border-t-neutral-300 mt-3.5"
          />
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              Member ID
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="accountId"
              value=""
            />
          </label>
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              Email
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="emailAddress"
              value=""
            />
          </label>
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              Postcode
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="postCode"
              value=""
            />
          </label>
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              Phone
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="phone"
              value=""
            />
          </label>
        </fieldset>
        <fieldset
          class="mb-2 w-full"
        >
          <hr
            class="border-t-neutral-300 mt-4.5"
          />
          <label
            class="-mt-4.5 relative cursor-pointer"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-stone-600"
            >
              Venture
            </p>
            <select
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full appearance-none cursor-pointer"
              name="ventures"
            />
            <svg
              class="absolute right-2 top-1/2 translate-y-0.5 pointer-events-none"
              data-file-name="SvgChevronDown"
            />
          </label>
        </fieldset>
        <fieldset
          class="mb-4"
        >
          <hr
            class="border-t-neutral-300 mt-3.5"
          />
          <label
            class="-mt-3.5 cursor-pointer"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-stone-600"
            >
              Screen name
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-pointer"
              data-test="screen-name-input"
              name="screenName"
              value="test1"
            />
          </label>
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              Last name
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="surname"
              value=""
            />
          </label>
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              First name
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="firstName"
              value=""
            />
          </label>
          <label
            class="-mt-3.5 cursor-not-allowed"
          >
            <p
              class="inline-block bg-white p-1 translate-x-3.5 translate-y-3.5 text-neutral-100"
            >
              DOB
            </p>
            <input
              class="bg-white block py-2.5 px-3 border border-neutral-400 rounded-lg w-full cursor-not-allowed border-neutral-100"
              disabled=""
              name="dateOfBirth"
              value=""
            />
          </label>
        </fieldset>
        <button
          class="uppercase bg-primary-500 text-white w-full p-3.5 rounded-lg"
          type="submit"
        >
          Search
        </button>
      </form>
    </section>
    <section
      class="flex-1 p-4 flex justify-center items-center"
    >
      <span
        style="display: inherit; position: relative; width: 100px; height: 4px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box;"
      >
        <span
          style="position: absolute; height: 4px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
        />
        <span
          style="position: absolute; height: 4px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
        />
      </span>
    </section>
  </main>
</DocumentFragment>
`;
